src/main/kotlin/uk/gov/dluhc/printapi/service/PrintService.kt

```diff
@@ -3,20 +3,34 @@ package uk.gov.dluhc.printapi.service
 import org.springframework.stereotype.Service
 import uk.gov.dluhc.printapi.database.repository.CertificateRepository
 import uk.gov.dluhc.printapi.mapper.CertificateMapper
+import uk.gov.dluhc.printapi.mapper.SourceTypeMapper
 import uk.gov.dluhc.printapi.messaging.models.SendApplicationToPrintMessage
 import javax.transaction.Transactional
 import uk.gov.dluhc.printapi.client.ElectoralRegistrationOfficeManagementApiClient as EroClient
 
 @Service
 class PrintService(
     private val eroClient: EroClient,
+    private val sourceTypeMapper: SourceTypeMapper,
     private val certificateMapper: CertificateMapper,
     private val certificateRepository: CertificateRepository
 ) {
     @Transactional
     fun savePrintMessage(message: SendApplicationToPrintMessage) {
         val ero = eroClient.getEro(message.gssCode!!)
-        val certificate = certificateMapper.toCertificate(message, ero)
+
+        var certificate = certificateRepository.findByGssCodeInAndSourceTypeAndSourceReference(
```

Comment:
It will be better to search using srcRef first. We did something similar in RCa (while deleting) so it may not be relevant. We also created an index on srcRef

Reply:
This is the same look up as used in the `CertificateFinderService` - possibly could have used that here and handled the `CertificateNotFoundException` exception.

There is already an existing index for this query:
```
        <createIndex indexName="source_details_inx" tableName="certificate" unique="true">
            <column name="gss_code"/>
            <column name="source_type"/>
            <column name="source_reference"/>
        </createIndex>
    </changeSet>
```